File: src/main/java/qmech/lib/tileentity/syncable/ISyncableObject.java
package qmech.lib.tileentity.syncable;

import net.minecraft.nbt.NBTTagCompound;

public interface ISyncableObject {
    /*

	public void readFromStream(DataInput stream) throws IOException;

	public void writeToStream(DataOutput stream, boolean fullData) throws IOException;
	*/

    public void writeToNBT(NBTTagCompound nbt, String name);

    public void readFromNBT(NBTTagCompound nbt, String name);

    public void markDirty();

    public boolean isDirty();

    public void markClean();
}

File: src/main/scala/geek/lawsof/physics/lib/block/nbt/ISyncableObject.scala
package geek.lawsof.physics.lib.block.nbt

import net.minecraft.nbt.NBTTagCompound

/**
 * Created by anshuman on 28-05-2014.
 */
trait ISyncableObject {
  def writeToNBT(nbt: NBTTagCompound, name: String)

  def readFromNBT(nbt: NBTTagCompound, name: String)

  var dirty = false

  def markDirty() = dirty = true

  def markClean() = dirty = false

  def value()
}
